src/libcore/mem/maybe_uninit.rs

```diff
@@ -258,6 +258,37 @@ impl<T> MaybeUninit<T> {
         MaybeUninit { uninit: () }
     }
 
+    /// Create a new array of `MaybeUninit<T>` items, in an uninitialized state.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(maybe_uninit_uninit_array, maybe_uninit_extra, maybe_uninit_slice_assume_init)]
+    /// use std::mem::MaybeUninit;
+    ///
+    /// let input = b"Foo";
+    /// let f = u8::to_ascii_uppercase;
+    ///
+    /// let mut buffer: [MaybeUninit<u8>; 32] = MaybeUninit::uninit_array();
+    /// let vec;
+    /// let output = if let Some(buffer) = buffer.get_mut(..input.len()) {
+    ///     buffer.iter_mut().zip(input).for_each(|(a, b)| { a.write(f(b)); });
+    ///     unsafe { MaybeUninit::slice_get_ref(buffer) }
+    /// } else {
+    ///     vec = input.iter().map(f).collect::<Vec<u8>>();
+    ///     &vec
+    /// };
+    ///
+    /// assert_eq!(output, b"FOO");
+    /// ```
+    #[unstable(feature = "maybe_uninit_uninit_array", issue = "0")]
+    #[inline(always)]
+    pub fn uninit_array<const LEN: usize>() -> [Self; LEN] {
+        unsafe {
+            MaybeUninit::<[MaybeUninit<T>; LEN]>::uninit().assume_init()
```

Comment:
This is a bit of a bummer. We should be able to use:
```rust
[Self::UNINIT; LEN]
```
here were it not for the fact that we would get *"error: array lengths can't depend on generic parameters"* (solvable by e.g. allowing `_` as the array length -- https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=28015a7e1c5856eef0ee32f4d53206a5).

Can you leave a FIXME for now?